File: Tests/Extensions/ContainsTests.fs
namespace Tests.Extensions

open System;
open Stringier.Extensions;
open Xunit;

module ContainsTests =
    [<Fact>]
    let ``contains 'a' "apple"`` () =
        Assert.True(contains 'a' "apple")

    [<Fact>]
    let ``contains 'a' "hello"`` () =
        Assert.False(contains 'a' "hello")

    [<Fact>]
    let ``contains 'g' ["ab";"cd";"ef";"gh"]`` () =
        Assert.True(contains 'g' ["ab";"cd";"ef";"gh"])

    [<Fact>]
    let ``contains 'i' ["ab";"cd";"ef";"gh"]`` () =
        Assert.False(contains 'i' ["ab";"cd";"ef";"gh"])

    [<Fact>]
    let ``contains 'g' [|"ab";"cd";"ef";"gh"|]`` () =
        Assert.True(contains 'g' [|"ab";"cd";"ef";"gh"|])

    [<Fact>]
    let ``contains 'i' [|"ab";"cd";"ef";"gh"|]`` () =
        Assert.False(contains 'i' [|"ab";"cd";"ef";"gh"|])

    [<Fact>]
    let ``contains "hello" "hello world"`` () =
        Assert.True(contains "hello" "hello world")

    [<Fact>]
    let ``contains "hello" "goodnight everyone"`` () =
        Assert.False(contains "hello" "goodnight everyone")

    [<Fact>]
    let ``contains "cd" ["ab";"cd";"ef";"gh"]`` ( ) =
        Assert.True(contains "cd" ["ab";"cd";"ef";"gh"])

    [<Fact>]
    let ``contains "de" ["ab";"cd";"ef";"gh"]`` () =
        Assert.False(contains "de" ["ab";"cd";"ef";"gh"])

    [<Fact>]
    let ``contains "cd" [|"ab";"cd";"ef";"gh"|]`` ( ) =
        Assert.True(contains "cd" ["ab";"cd";"ef";"gh"])

    [<Fact>]
    let ``contains "de" [|"ab";"cd";"ef";"gh"|]`` () =
        Assert.False(contains "de" ["ab";"cd";"ef";"gh"])

File: Tests/ContainsTests.cs
using System;
using System.Collections.Generic;
using Stringier;
using Xunit;

namespace Tests {
	public class ContainsTests {
		[Theory]
		[InlineData("apple", 'a', true)]
		[InlineData("hello", 'a', false)]
		public void String_Contains_Char(String source, Char value, Boolean expected) => Assert.Equal(expected, source.Contains(value));

		[Theory]
		[InlineData("hello world", "hello", true)]
		[InlineData("goodnight everyone", "hello", false)]
		public void String_Contains_String(String source, String value, Boolean expected) => Assert.Equal(expected, source.Contains(value));

		[Theory]
		[InlineData(new[] { "ab","cd","ef","gh"}, 'g', true)]
		[InlineData(new[] { "ab", "cd", "ef", "gh" }, 'i', false)]
		public void Array_Contains_Char(String[] source, Char value, Boolean expected) => Assert.Equal(expected, source.Contains(value));

		[Theory]
		[InlineData(new[] { "ab","cd","ef","gh"}, "cd", true)]
		[InlineData(new[] { "ab", "cd", "ef", "gh" }, "de", false)]
		public void Array_Contains_String(String[] source, String value, Boolean expected) => Assert.Equal(expected, source.Contains(value));

		[Theory]
		[InlineData(new[] { "ab", "cd", "ef", "gh" }, 'g', true)]
		[InlineData(new[] { "ab", "cd", "ef", "gh" }, 'i', false)]
		public void List_Contains_Char(String[] source, Char value, Boolean expected) => Assert.Equal(expected, new List<String>(source).Contains(value));

		[Theory]
		[InlineData(new[] { "ab", "cd", "ef", "gh" }, "cd", true)]
		[InlineData(new[] { "ab", "cd", "ef", "gh" }, "de", false)]
		public void List_Contains_String(String[] source, String value, Boolean expected) => Assert.Equal(expected, new List<String>(source).Contains(value));
	}
}
